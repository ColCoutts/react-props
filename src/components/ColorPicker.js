import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './style.modules.css';

export default class ColorPicker extends PureComponent {
  static propTypes = {
    colors: PropTypes.array.isRequired
  }

  colors = ['red', 'blue', 'green'];

clickHandlerRed = () => {
  console.log(this.props.colors[0]);
}

clickHandlerBlue = () => {
  console.log(this.props.colors[1]);
}

clickHandlerGreen = () => {
  console.log(this.props.colors[2]);
}

render() {
  return (
    <div> 
      <button className={styles.red} onClick={this.clickHandlerRed}>{'red'}</button>
      <button className={styles.blue} onClick={this.clickHandlerBlue}>{'blue'}</button>
      <button className={styles.green} onClick={this.clickHandlerGreen}>{'green'}</button>
    </div>
  );
}
}
