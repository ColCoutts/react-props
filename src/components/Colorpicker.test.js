import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('Snapshot tests', () => {
  it('colorpicker snapshot', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });
});
