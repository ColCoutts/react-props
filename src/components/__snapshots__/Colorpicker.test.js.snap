// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests colorpicker snapshot 1`] = `
<div>
  <button
    className="red"
    onClick={[Function]}
  >
    red
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    blue
  </button>
  <button
    className="green"
    onClick={[Function]}
  >
    green
  </button>
</div>
`;
